rm-portal/src/main/java/org/ow2/proactive_grid_cloud_portal/rm/client/monitoring/views/FileSystemView.java

```diff
@@ -88,40 +84,31 @@ public void onSuccess(String result) {
-
-                                     Options opts = Options.create();
-                                     opts.setLegend(LegendPosition.NONE);
-                                     opts.setColors("#fcaf3e",
-                                                    "#3a668d",
-                                                    "#35a849",
-                                                    "#fcaf3e",
-                                                    "#24c1ff",
-                                                    "#1e4ed7",
-                                                    "#ef2929",
-                                                    "#000000");
-
                                      JSONObject object = controller.parseJSON(result).isObject();
                                      if (object != null) {
 
                                          for (String disk : object.keySet()) {
 
                                              HLayout diskLayout = new HLayout();
 
-                                             DataTable pieData = DataTable.create();
-                                             pieData.addColumn(ColumnType.STRING, "Type");
-                                             pieData.addColumn(ColumnType.NUMBER, "Bytes");
-
                                              DetailViewer details = new DetailViewer();
                                              DetailViewerRecord dv = new DetailViewerRecord();
                                              DetailViewerField[] fields = new DetailViewerField[attrs.size()];
                                              for (int i = 0; i < fields.length; i++) {
                                                  fields[i] = new DetailViewerField(attrs.get(i));
                                              }
                                              details.setFields(fields);
+                                             details.setData(new DetailViewerRecord[] { dv });
+                                             details.setWidth("50%");
 
+                                             PieChart pie = new PieChart();
+                                             PieDataset dataset = pie.newDataset();
+                                             dataset.setBackgroundColor(new String[] { "#fcaf3e", "#3a668d", "#35a849",
```

Comment:
Do you think you can create the enum with colors code and name? You use the same color codes in many places. It will be easier to change in future.

Reply:
I will think about it how to do it...